docs/source/models.rst

```diff
@@ -3769,30 +3769,65 @@ back to ``numpy ndarray`` type as required by ``sktime`` inference API.
-You can utilize this functionality by passing a sample input, as demonstrated in the example code below.
-Additionally, this API is helpful for validating the model dependecies. 
-For more details, refer to  :ref:`Validating Environment for Prediction <validating-environment-for-prediction>`.
+After logging your model with MLflow Tracking, it is highly recommended to validate the model locally before deploying it to production.
+The :py:func:`mlflow.models.predict` API provides a convenient way to test your model in a virtual environment, offering isolated execution and several advantages:
+
+* Model dependencies validation: The API helps ensure that the dependencies logged with the model are correct and sufficient by executing the model with an input example in a virtual environment.
+  For more details, refer to :ref:`Validating Environment for Prediction <validating-environment-for-prediction>`.
+* Input data validation: The API can be used to validate the input data interacts with the model as expected by simulating the same data processing during model serving.
+  Ensure that the input data is a valid example that aligns with the pyfunc model’s predict function requirements.
+* Extra environment variables validation: By specifying the `extra_envs` parameter, you can test whether additional environment variables are needed for the model to run successfully.
+  Note that all existing environment variables in `os.environ` are automatically passed into the virtual environment.
 
 .. code-block:: python
 
     import mlflow
 
 
     class MyModel(mlflow.pyfunc.PythonModel):
-        def predict(self, context, model_input):
-            return len(model_input)
+        def predict(self, context, model_input, params=None):
+            return model_input
 
 
-    model_info = mlflow.pyfunc.log_model(
-        artifact_path="model",
-        python_model=MyModel(),
-    )
+    with mlflow.start_run():
+        model_info = mlflow.pyfunc.log_model(
+            "model",
+            python_model=MyModel(),
+            input_example=["a", "b", "c"],
+        )
 
     mlflow.models.predict(
         model_uri=model_info.model_uri,
-        input_data=[1, 2],
+        input_data=["a", "b", "c"],
+        pip_requirements_override=["..."],
+        extra_envs={"MY_ENV_VAR": "my_value"},
+    )
+
+Different environment managers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The :py:func:`mlflow.models.predict` API supports different environment managers to create the virtual environment for prediction. The following environment managers are supported:
+
+* `virtualenv <https://virtualenv.pypa.io/en/latest/>`_: the default environment manager which creates a virtual environment using `virtualenv` and `pyenv`.
+* `uv <https://docs.astral.sh/uv/>`_: uses `uv` to create a virtual environment. **This is an experimental feature since MLflow 2.20.0.**
+* `conda <https://docs.conda.io/projects/conda/>`_: uses conda to create environment.
```

Comment:
```suggestion
* `virtualenv <https://virtualenv.pypa.io/en/latest/>`_: The default environment manager.
* `uv <https://docs.astral.sh/uv/>`_:  An **extremely fast** environment manager written in Rust. This is an experimental feature since MLflow 2.20.0.
* `conda <https://docs.conda.io/projects/conda/>`_
```

nit: It sounds a bit verbose to say "A is a env manager that uses A"